packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm

```diff
@@ -611,54 +611,88 @@ static RCTBorderStyle RCTBorderStyleFromBorderStyle(BorderStyle borderStyle)
-      return [NSCursor contextualMenuCursor];
+      return RCTCursorAlias;
+    case Cursor::AllScroll:
+      return RCTCursorAllScroll;
+    case Cursor::Cell:
+      return RCTCursorCell;
+    case Cursor::ColResize:
+      return RCTCursorColResize;
+    case Cursor::ContextMenu:
+      return RCTCursorContextMenu;
     case Cursor::Copy:
-      return [NSCursor dragCopyCursor];
+      return RCTCursorCopy;
     case Cursor::Crosshair:
-      return [NSCursor crosshairCursor];
+      return RCTCursorCrosshair;
     case Cursor::Default:
-      return [NSCursor arrowCursor];
-    case Cursor::DisappearingItem:
-      return [NSCursor disappearingItemCursor];
-    case Cursor::EastResize:
-      return [NSCursor resizeRightCursor];
+      return RCTCursorDefault;
+    case Cursor::EResize:
+      return RCTCursorEResize;
+    case Cursor::EWResize:
+      return RCTCursorEWResize;
     case Cursor::Grab:
-      return [NSCursor openHandCursor];
+      return RCTCursorGrab;
     case Cursor::Grabbing:
-      return [NSCursor closedHandCursor];
-    case Cursor::NorthResize:
-      return [NSCursor resizeUpCursor];
+      return RCTCursorGrabbing;
+    case Cursor::Help:
+      return RCTCursorHelp;
+    case Cursor::Move:
+      return RCTCursorMove;
+    case Cursor::NEResize:
+      return RCTCursorNEResize;
+    case Cursor::NESWResize:
+      return RCTCursorNESWResize;
+    case Cursor::NResize:
+      return RCTCursorNResize;
+    case Cursor::NSResize:
+      return RCTCursorNSResize;
+    case Cursor::NWResize:
+      return RCTCursorNWResize;
+    case Cursor::NWSEResize:
+      return RCTCursorNWSEResize;
     case Cursor::NoDrop:
-      return [NSCursor operationNotAllowedCursor];
+      return RCTCursorNoDrop;
+    case Cursor::None:
+      return RCTCursorNone;
     case Cursor::NotAllowed:
-      return [NSCursor operationNotAllowedCursor];
+      return RCTCursorNotAllowed;
     case Cursor::Pointer:
-      return [NSCursor pointingHandCursor];
+      return RCTCursorPointer;
+    case Cursor::Progress:
+      return RCTCursorProgress;
     case Cursor::RowResize:
-      return [NSCursor resizeUpDownCursor];
-    case Cursor::SouthResize:
-      return [NSCursor resizeDownCursor];
+      return RCTCursorRowResize;
+    case Cursor::SResize:
+      return RCTCursorSResize;
+    case Cursor::SEResize:
+      return RCTCursorSEResize;
+    case Cursor::SWResize:
+      return RCTCursorSWResize;
     case Cursor::Text:
-      return [NSCursor IBeamCursor];
+      return RCTCursorText;
+    case Cursor::Url:
+      return RCTCursorUrl;
     case Cursor::VerticalText:
-      return [NSCursor IBeamCursorForVerticalLayout];
-    case Cursor::WestResize:
-      return [NSCursor resizeLeftCursor];
+      return RCTCursorVerticalText;
+    case Cursor::WResize:
+      return RCTCursorWResize;
+    case Cursor::Wait:
+      return RCTCursorWait;
+    case Cursor::ZoomIn:
+      return RCTCursorZoomIn;
+    case Cursor::ZoomOut:
```

Comment:
Is this switch exhaustive? Should we have a default case that throws?

Reply:
It is exhaustive, and Xcode should complain / not compile if it isn't I'm pretty sure. I'll double check by removing an entry

Reply:
It will throw a warning. I'd rather keep the warning (and eventually maybe add a build setting to turn it into an error) rather than suppress it with a default statement.

Reply:
Yeah, let's look at making it an error later.